evm/vm_state.py

```diff
@@ -324,3 +318,19 @@ def get_uncle_reward(block_number, uncle):
     @classmethod
     def get_nephew_reward(cls):
         raise NotImplementedError("Must be implemented by subclasses")
+
+
+class BaseTransactionExecutor:
+    def execute_transaction(self, transaction):
+        message = self.run_pre_computation(transaction)
+        computation = self.run_computation(transaction, message)
+        return self.run_post_computation(transaction, computation)
+
+    def run_pre_computation(self, transaction):
```

Comment:
can you update these to be decorated with `abc.abstractmethod` as was recently added elsewhere in #467